src/battle_anim_80CA710.c

```diff
@@ -0,0 +1,4381 @@
+            b = sub_8077ABC(gBattleAnimEnemyMonIndex, 1);
+            break;
+        }
+    }
+    sprite->pos1.x = a;
+    sprite->pos1.y = b;
+    if(GetBankSide(gBattleAnimEnemyMonIndex) == 0)
+        sprite->pos1.y += 8;
+    sprite->callback = sub_80CCB00;
+    if(gBattleAnimArgs[2] == 0)
+        sprite->pos1.x += gBattleAnimArgs[0];
+    else
+    {
+        sprite->pos1.x -= gBattleAnimArgs[0];
+        sprite->hFlip = 1;
+    }
+    sprite->pos1.y += gBattleAnimArgs[1];
+    sprite->data1 -= 0x400;
+    sprite->data2 += 0x400;
+    sprite->data5 = gBattleAnimArgs[2];
+    if(sprite->data5 == 1)
+        sprite->data1 = -sprite->data1;
+}
+
+void sub_80CCB00(struct Sprite* sprite)
+{
+    sprite->data3 += sprite->data1;
+    sprite->data4 += sprite->data2;
+    if(sprite->data5 == 0)
+        sprite->data1 += 0x18;
+    else
+        sprite->data1 -= 0x18;
+    sprite->data2 -= 0x18;
+    sprite->pos2.x = sprite->data3 >> 8;
+    sprite->pos2.y = sprite->data4 >> 8;
+    sprite->data0++;
+    if(sprite->data0 == 20)
+    {
+        oamt_set_x3A_32(sprite, move_anim_8072740);
+        sprite->data0 = 3;
+        sprite->callback = sub_80782D8;
+    }
+}
+
+void unref_sub_80CCB6C(struct Sprite* sprite)
+{
+    if(sprite->data2 > 1)
+    {
+        if(sprite->data3 & 1)
+        {
+            sprite->invisible = 0;
+            gSprites[sprite->data0].invisible = 0;
+            gSprites[sprite->data1].invisible = 0;
+        }
+        else
+        {
+            sprite->invisible = 1;
+            gSprites[sprite->data0].invisible = 1;
+            gSprites[sprite->data1].invisible = 1;
+        }
+        sprite->data2 = 0;
+        sprite->data3++;
+    }
+    else
+    {
+        sprite->data2++;
+    }
+    if(sprite->data3 == 10)
+    {
+        DestroySprite(&gSprites[sprite->data0]);
+        DestroySprite(&gSprites[sprite->data1]);
+        move_anim_8072740(sprite);
+    }
+}
+
+void sub_80CCC50(struct Sprite* sprite)
+{
+    sprite->data0 = gBattleAnimArgs[2];
+    if(GetBankSide(gBattleAnimPlayerMonIndex) != 0)
+        sprite->pos1.x -= gBattleAnimArgs[0];
+    else
+        sprite->pos1.x += gBattleAnimArgs[0];
+    StartSpriteAnim(sprite, gBattleAnimArgs[5]);
+    sprite->data1 = -gBattleAnimArgs[3];
+    sprite->pos1.y += gBattleAnimArgs[1];
+    sprite->data3 = gBattleAnimArgs[4];
+    sprite->callback = sub_80CCCB4;
+    sub_80CCCB4(sprite);
+}
+
+void sub_80CCCB4(struct Sprite* sprite)
+{
+    sprite->pos2.x = Cos(sprite->data0, 100);
+    sprite->pos2.y = Sin(sprite->data0, 20);
+    if(sprite->data0 <= 0x7F)
+        sprite->subpriority = 0;
+    else
+        sprite->subpriority = 14;
+    sprite->data0 = (sprite->data0 + sprite->data1) & 0xFF;
+    sprite->data5 += 0x82;
+    sprite->pos2.y += sprite->data5 >> 8;
+    sprite->data2++;
+    if(sprite->data2 == sprite->data3)
+        move_anim_8072740(sprite);
+}
+
+void sub_80CCD24(struct Sprite* sprite)
+{
+    if(IsContest() != 0)
```

Comment:
`IsContest()` returns a boolean, so it's more readable to do `if (IsContest())`.